### ISSUE TITLE
Use threads for better performance

### ISSUE THREAD

**Issue by kinex** [builttoroam/flutter_plugins#181]
I noticed this plugin is doing its work in UI thread. I can already see some slowness in my app UI. Could you please consider applying threading for better performance. This is also recommended at least in related Android [documentation](https://developer.android.com/reference/android/provider/CalendarContract.Instances#query(android.content.ContentResolver,%20java.lang.String%5B%5D,%20long,%20long)) .

Threading is very simple in Kotlin and in Swift. If you need help, here are some code templates (this is how I do it in my own plugins):

Kotlin:
```kotlin
val uiScope = CoroutineScope(Dispatchers.Main)
uiScope.launch {
    // notice placement of try/catch
    try {
        withContext(Dispatchers.IO) {
            // do heavy work here
        }
        // this must be called on UI thread
        result.success(true)
    } catch (e: Exception) {
        // this must be called on UI thread
        result.error("error", e.localizedMessage, e.toString())
    }
}
```

Swift:
```swift
// you may also consider other options than .userInitiated
DispatchQueue.global(qos: .userInitiated).async {
    do {
        // do heavy work here
        
        DispatchQueue.main.async {
            // this must be called on UI thread
            result(true)
        }
    } catch {
        DispatchQueue.main.async {
            // this must be called on UI thread
            result(FlutterError(code: "ERROR",
                                message: error.localizedDescription,
                                details: nil))
        }
    }
}
```

**Comment by Brett09** [comment]
Thanks for the suggestion & examples. Will look into them and apply them when I get a chance.

**Comment by intkhabahmed** [comment]
@Brett09 what is the progress in this, it is causing my app to skip frames, please resolve this asap.

**Comment by dungtang93** [comment]
> @Brett09 what is the progress in this, it is causing my app to skip frames, please resolve this asap.

https://github.com/builttoroam/device_calendar/pull/222#issuecomment-643838348

Android only, as I don't have an iOS phone to code & test this out, help wanted

**Comment by imhafeez** [comment]
`createOrUpdateEvent` is also on main thread. it causes frame skips as well. Please change it to background.

**Comment by dungtang93** [comment]
@imhafeez Why are you experiencing frame skip in `createOrUpdateEvent`? are you creating a lot of events at once?

**Comment by imhafeez** [comment]
@dungtang93 yes. actually I was syncing all the events from server to my local calendar. any solution for that?

**Comment by thomassth** [comment]
Seems that we have threading on Android but not on iOS.

Or can someone more proficient with kotlin / swift make a checklist on what's done and not done?
